snow/networking/router/chain_router.go

```diff
@@ -273,7 +280,6 @@
 	sr.lock.RLock()
 	defer sr.lock.RUnlock()
```

Comment:
Should this be at the end of the func ?

Reply:
When possible with locking, I like to keep the `Unlock` in a defer so that `return`s mid function still work. The performance impact of a defer statement is pretty negligible in the newer go versions (Unless you are calling them in a for loop).